src/js/core/widget/core/DatePicker.js

```diff
@@ -0,0 +1,1844 @@
+ */
+/*
+ * @example
+ * <div class="ui-datepicker" id="pickers"></div>
+ *
+ * id="pickers" should be fixed.
+ *
+ * @since 6.0
+ * @class ns.widget.core.DatePicker
+ * @extends ns.widget.BaseWidget
+ * @author Dohyung Lim <delight.lim@samsung.com>
+ */
+(function (window, document, ns) {
+	"use strict";
+	//>>excludeStart("tauBuildExclude", pragmas.tauBuildExclude);
+	define(
+		[
+			"../../engine",
+			"../BaseWidget"
+		],
+		function () {
+			//>>excludeEnd("tauBuildExclude");
+			var utilsObject = ns.util.object,
+
+				DatePicker = function () {
+					this.options = utilsObject.merge({}, DatePicker.defaults);
+					this.observer = null;
+				},
+
+				defaults = {
+					value: false, // false is past selectable, true is past selection not possible.
+					type: "calendar" // calendar or wheel
+				},
+
+				BaseWidget = ns.widget.BaseWidget,
+				prototype = new BaseWidget();
+
+			DatePicker.prototype = prototype;
+			DatePicker.defaults = defaults;
+
+			prototype._init = function (element) {
+                var TDP;
+
+				if (typeof (TDP) == "undefined") {
+					TDP = {
+						newId: function () {
+							return this.lastId++;
+						}
+					};
+					TDP.isIE10 = (navigator.userAgent.indexOf("MSIE 10.0") >= 0 && navigator.userAgent.indexOf("Trident") >= 0 && window.navigator.pointerEnabled != true);
+					TDP.isIE11 = (navigator.userAgent.indexOf("Trident") >= 0 && window.navigator.pointerEnabled == true);
+					TDP.isEdge = (navigator.userAgent.indexOf("Edge") >= 0);
+					TDP.isOpera = (navigator.userAgent.indexOf("Opera") >= 0);
+					TDP.isChrome = (navigator.userAgent.indexOf("Chrome") >= 0) && !TDP.isEdge;
+					TDP.isKHTML = (navigator.userAgent.indexOf("Safari") >= 0 || navigator.userAgent.indexOf("Konqueror") >= 0) && !TDP.isEdge;
+					TDP.isFF = (navigator.userAgent.indexOf("Firefox") >= 0);
+					TDP.isIPad = (navigator.userAgent.search(/iPad/gi) >= 0);
+					TDP.dnd = {
+						evs: {},
+						// eslint-disable-next-line camelcase
+						p_en: ((TDP.isIE || TDP.isEdge) && (window.navigator.pointerEnabled || window.navigator.msPointerEnabled)),
+						_mTouch: function (a) {
+							return (TDP.isIE10 && a.pointerType == a.MSPOINTER_TYPE_MOUSE || TDP.isIE11 && a.pointerType == "mouse" || TDP.isEdge && a.pointerType == "mouse")
+						},
+						_touchOn: function (a) {
+							if (a == null) {
+								a = document.body;
+							}
+							a.style.touchAction = a.style.msTouchAction = "";
+							a = null;
+						},
+						_touchOff: function (a) {
+							if (a == null) {
+								a = document.body;
+							}
+							a.style.touchAction = a.style.msTouchAction = "none";
+							a = null;
+						}
+					};
+					if (window.navigator.pointerEnabled == true) {
+						TDP.dnd.evs = {
+							start: "pointerdown",
+							move: "pointermove",
+							end: "pointerup"
+                        }
+                    } else {
+                        if (window.navigator.msPointerEnabled == true) {
```

Comment:
Do we need to consider the variable of the "msPointerEnabled"?

Reply:
no. this condition not needed.
I removed.

done.